tests/HashMapProperties.hs

```diff
@@ -290,6 +293,12 @@ pFoldrWithKey = (sortByKey . M.foldrWithKey f []) `eq`
                 (sortByKey . HM.foldrWithKey f [])
   where f k v z = (k, v) : z
 
+#if MIN_VERSION_base(4,10,0)
+pBifoldr :: [(Int, Int)] -> Bool
```

Comment:
Could you add a comment what we're testing here?

What's the  `sort` for BTW? Can't we expect that `foldrWithKey` and `bifoldr` traverse things in the same order?

Reply:
The `sort` is because there's a `Map` on the left and a `HashMap` on the right. I would have rather written `bifoldr` on both of them, but that would add a dependency on the latest, still-unreleased version of `containers`.

Reply:
Hm, I still don't quite understand what this tests. Just that `bifoldr` covers all the keys and values?

The relation to `Data.Map` is very non-obvious to me.

Wouldn't it be more interesting to test that all keys and values are accumulated in the expected order? Something like

```
concat (zipWith (\k v -> [k, v]) (keys m) (Foldable.toList m)) == bifoldr …
```

Reply:
Or, simpler

`HM.toList == bifoldr ...`

Reply:
How is it now?

Reply:
Much better! I didn't have the correct type of `bifoldr` in mind in my suggestions above.

Another possible improvement would be to use keys prone to hash collisions, but I'm not sure whether that's asking for too much.